pkgs/by-name/ib/ibm-plex/package.nix

```
@@ -1,28 +1,55 @@
-{ lib, stdenvNoCC, fetchzip }:
+{
+  lib,
+  stdenvNoCC,
+  fetchzip,
+  families ? [ ],
+}:
+let
+  version = import ./version.nix;
+  availableFamilies = import ./hashes.nix;
 
-stdenvNoCC.mkDerivation rec {
+  availableFamilyNames = builtins.attrNames availableFamilies;
+  selectedFamilies = if (families == [ ]) then availableFamilyNames else families;
+
+  unknownFamilies = lib.subtractLists availableFamilyNames families;
+in
+assert lib.assertMsg (unknownFamilies == [ ]) "Unknown font(s): ${toString unknownFamilies}";
+stdenvNoCC.mkDerivation {
   pname = "ibm-plex";
-  version = "6.4.0";
+  inherit version;
 
-  src = fetchzip {
-    url = "https://github.com/IBM/plex/releases/download/v${version}/OpenType.zip";
-    hash = "sha256-/aR3bu03VxenSPed6EqrGoPjWCcKT//MVtb9OC8tSRs=";
-  };
+  srcs = builtins.map (
+    family:
+    fetchzip {
+      url = "https://github.com/IBM/plex/releases/download/%40ibm%2Fplex-${family}%40${version}/ibm-plex-${family}.zip";
+      hash = availableFamilies.${family};
+    }
+  ) selectedFamilies;
+
+  dontUnpack = true;
+  sourceRoot = ".";
 
   installPhase = ''
     runHook preInstall
-
-    install -Dm644 */*.otf IBM-Plex-Sans-JP/unhinted/* -t $out/share/fonts/opentype
-
+    for src in $srcs; do
+      for f in $src/fonts/complete/otf/{hinted/,}*.otf; do
+        install -Dm644 $f -t $out/share/fonts/opentype
+      done
+    done
```

```suggestion
  for f in $srcs/*/fonts/complete/otf/{hinted/,}*.otf; do
    install -Dm644 $f -t $out/share/fonts/opentype
  done
```
Shouldn't this work without nesting?

---

This doesn't work because `srcs` is a space-delimited list of the derivation paths and not a directory containing the derivations.

---

Kinda forgot about that...

`find $srcs -type f -name '*.otf' -exec install -Dm644 {} -t $out/share/fonts/opentype {} \;` should work as an alternative